src/llfs/page_allocator_state.cpp

```diff
@@ -420,18 +472,25 @@ void PageAllocatorState::learn(slot_offset_type index_slot, const PackedPageAllo
     const PageId page_id{delta.page_id.value()};
     const page_id_int physical_page = this->page_ids_.get_physical_page(page_id);
     PageAllocatorRefCount* const obj = &this->page_ref_counts_[physical_page];
-    this->learn_ref_count_delta(delta, obj, metrics);
-    this->set_last_update(obj, index_slot);
+
+    obj->set_last_modified_by(txn.user_index);
+
+    this->learn_ref_count_delta(slot_offset, delta, obj, metrics);
+
+    if (delta.ref_count == kRefCount_1_to_0 || !obj->PageAllocatorLRUHook::is_linked()) {
```

Comment:
The fix to #25

Reply:
May be we can think of collapsing all (past) updates into the new one so that we can remove our past dependencies. Alternatively, we can always write a whole record which is fully qualified to generate the latest value of the ref count.